File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    
    students = [1] * n
    
    for item in lost:
        students[item - 1] -= 1
    
    for item in reserve:
        students[item - 1] += 1
    
    for i in range(n):
        if students[i] == 0:
            # 왼쪽 확인
            if i - 1 >= 0 and students[i - 1] == 2:
                students[i - 1] = 1
                students[i] = 1
            # 오른쪽 확인
            elif i + 1 < n and students[i + 1] == 2:
                students[i + 1] = 1
                students[i] = 1
    
    for i in students:
        if i >= 1:
            answer += 1
    
    return answer

File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    const checkArray = Array(n).fill(1);
    lost.forEach((num) => checkArray[num - 1]--);
    reserve.forEach((num) => checkArray[num - 1]++);
    
    checkArray.forEach((num, i) => {
        if (checkArray[i] === 0) {
            // 앞번호 확인
            if (i !== 0 && checkArray[i - 1] === 2) {
                checkArray[i - 1]--;
                checkArray[i]++;
            } else if (i !== n - 1 && checkArray[i + 1] === 2) {
                checkArray[i + 1]--;
                checkArray[i]++;
            }
        }
    })

    return checkArray.filter((num) => num !== 0).length;
}